### ISSUE TITLE
ClrMD needs to throw a different exception when asked to load a Portable PDB

### ISSUE THREAD

**Issue by skybook888** [microsoft/clrmd#215]
a simplly code
```
using System;
using Microsoft.Diagnostics.Runtime;
using Microsoft.Diagnostics.Runtime.Utilities;
using Microsoft.Diagnostics.Runtime.Utilities.Pdb;
namespace pdbreadertest
{
    class Program
    {
        static void Main(string[] args)
        {
           if (args[0]!=null )
           {
                   var pdbPath = args[0];
                   var reader = new PdbReader(pdbPath);
                Console.WriteLine(reader.Version);

           }
        }
    }
}
```
Build it (pdbreadertest is the project name)
```
dotnet build -o pub
```
just read the pdb file builder by it self
```
dotnet  pub/pdbreadertest.dll pub/pdbreadertest.pdb
```
reports 
```
Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Microsoft.Diagnostics.Runtime.Utilities.Pdb.BitAccess.ReadInt32(Int32& value)
   at Microsoft.Diagnostics.Runtime.Utilities.Pdb.BitAccess.ReadInt32(Int32[] values)
   at Microsoft.Diagnostics.Runtime.Utilities.Pdb.MsfDirectory..ctor(PdbStreamHelper reader, PdbFileHeader head, BitAccess bits)
   at Microsoft.Diagnostics.Runtime.Utilities.Pdb.PdbFile.LoadFunctions(Stream read, BitAccess bits, Boolean readAllStrings, Int32& ver, Int32& sig, Int32& age, Guid& guid, IEnumerable`1& sources)
   at Microsoft.Diagnostics.Runtime.Utilities.Pdb.PdbReader.Init(Stream pdbStream)
   at Microsoft.Diagnostics.Runtime.Utilities.Pdb.PdbReader..ctor(String fileName)
   at pdbreadertest.Program.Main(String[] args) in /pdbreadertest/Program.cs:line 14
```
The dotnet info 
```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.100
 Commit:    51868761f2

Runtime Environment:
 OS Name:     debian
 OS Version:  9
 OS Platform: Linux
 RID:         debian.9-x64
 Base Path:   /usr/share/dotnet/sdk/2.2.100/

Host (useful for support):
  Version: 2.2.0
  Commit:  1249f08fed

.NET Core SDKs installed:
  2.2.100 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by skybook888** [comment]
By the way，is there any way to read the portable pdbs?
my project is a dotnet core project running on linux.
I want to use clrmd to auto analysis the coredump files.
if PdbReader conld not support portable pdbs ,
how to find the line numbers?

**Comment by leculver** [comment]
@skybook888 I don't know the answer to your question.  I have asked other people on the CLR team to respond to this thread.  Thank you!

**Comment by leculver** [comment]
From @mikem8361: 

> You can use this code as a guide for using System.Reflection.Metadata to read portable PDBs.  This is the SOS code to load, get source/line info and local variable names:
> 
> https://github.com/dotnet/diagnostics/blob/master/src/SOS/SOS.NETCore/SymbolReader.cs
> 
> Here are some pending fixes/changes:
> 
> https://github.com/mikem8361/diagnostics/blob/installer/src/SOS/SOS.NETCore/SymbolReader.cs

**Comment by leculver** [comment]
PDB support is slated for removal in ClrMD 2.0, please see here:  https://github.com/microsoft/clrmd/blob/master/doc/v2_roadmap.md

I have decided not to fix this for 1.0 and instead just leave it as-is.
